x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/index.tsx

```diff
@@ -83,8 +84,12 @@ export class Waterfall extends Component<Props, State> {
   };
 
   public getFlyOut = () => {
-    const { currentItem } = this.state;
-    const { waterfall } = this.props;
+    const { waterfall, location, urlParams } = this.props;
+
+    const currentItem =
+      urlParams.waterfallItemId &&
+      waterfall.itemsById[urlParams.waterfallItemId];
```

Comment:
Maybe you answered this previously but do we need the check for `urlParams.waterfallItemId`? Since we have a check below (if(`!currentItem`)) I think this should be fine:

```js
const currentItem = waterfall.itemsById[urlParams.waterfallItemId];
```

Reply:
Blame typescript! :P
```
[ts] Type 'undefined' cannot be used as an index type.
```

Reply:
Ah! Didn't know that!

Reply:
That is actually weird. `undefined` is a perfectly valid index value:
```js
const obj = {undefined: 'foo'}
obj[undefined] // "foo"
```

Reply:
I know! I was annoyed too